ipywidgets/static/widgets/js/widget_color.js

```diff
@@ -0,0 +1,89 @@
+
+    /*
+     * From a valid html color (named color, 6-digits or 3-digits hex format)
+     * return a 6-digits hexadecimal color #rrggbb.
+     */
+    function color2hex(color) {
+        return named_colors[color.toLowerCase()] || rgb2hex(color);
+    };
+
+    function rgb2hex(rgb) {
+        if (rgb.search("rgb") == -1) {
+            return rgb;
+        } else {
+            rgb = rgb.match(/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+))?\)$/);
+            function hex(x) {
+                return ("0" + parseInt(x).toString(16)).slice(-2);
+            }
+            return "#" + hex(rgb[1]) + hex(rgb[2]) + hex(rgb[3]);
+        }
+    };
+
+    return {
+        ColorPicker: ColorPicker,
+        color2hex: color2hex,
```

Comment:
I'm confused, what makes color2hex > rgb2hex ?  Why export one and not the other?   Or instead, not export anything but the widget.